### ISSUE TITLE
Brands only showing up when "All" styles is selected

### ISSUE THREAD

**Issue by pascalminator** [verbb/icon-picker#97]
### Describe the bug

If I manually check some styles including "Brands" instead of selecting "All", the brands are not showing up in the search results of the icon field. I tried many different combinations, and the only way to get brands in the field is by checking "All".

Brands not showing up:
![image](https://github.com/user-attachments/assets/961e8910-2f81-4fb0-9fe7-aadbc7f3654b)

Brands showing up:
![image](https://github.com/user-attachments/assets/a35898cb-d1da-4ace-a75a-2802d56c2dc8)

### Steps to reproduce

1. Uncheck "All" in "Styles" and check some other styles, including "Brands"
2. Try to find a brand (Facebook, Instagram, wathever) in the field without any success.
3. Check "all" styles and try again. Now the brands are showing up. 

### Craft CMS version

5.4.7.1

### Plugin version

3.0.1

### Multi-site?

No

**Comment by engram-design** [comment]
Fixed for the next release. To get this early, run `composer require verbb/icon-picker:"dev-craft-5 as 3.0.1"`. Note that you'll need to re-save your Icon Set, as the "Brands" style had an incorrect value.

**Comment by pascalminator** [comment]
This is fixed if I check "Brands" and at least one other style but if I check only "Brands" and nothing else, there is an error:
![image](https://github.com/user-attachments/assets/29973173-adc8-4d6c-8c8c-f66c7c5a8017)

Getting this problem with Icon Picker 3.0.2.

**Comment by engram-design** [comment]
I'm not seeing that on my end, sorry. What are your icon set settings?

Anything in your logs?

![CleanShot 2024-11-21 at 11 38 23@2x](https://github.com/user-attachments/assets/a6335a29-d96d-445e-b533-5a57fd68f623)

**Comment by pascalminator** [comment]
My settings are the same as in your screenshot, but my kit use Font Awesome 6 Pro instead of 5 Free.

Here is what I found in the queue log:

2024-11-21 08:49:46 [queue.ERROR] [craft\queue\QueueLogBehavior::afterError]  [460] Generating icon cache for "brands" (attempt: 1, pid: 237732) - Error (time: 0.002s): verbb\iconpicker\iconsets\FontAwesome::getKit(): Return value must be of type array, null returned {"memory":2383672} 
2024-11-21 08:49:46 [queue.ERROR] [TypeError] TypeError: verbb\iconpicker\iconsets\FontAwesome::getKit(): Return value must be of type array, null returned in /var/www/html/vendor/verbb/icon-picker/src/iconsets/FontAwesome.php:298

**Comment by siebird** [comment]
I'm getting the same error today as well. The icons stopped working and I'm not seeing any kits to select from.

![CleanShot 2024-11-21 at 11 10 43](https://github.com/user-attachments/assets/4cd18039-61f4-4459-a748-945e81bf25e3)
 

```$_POST = []
2024-11-21 10:47:22 [ERROR] FontAwesome Kit API error: “error code: 502” /var/www/html/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:111
2024-11-21 10:47:22 [INFO] Request context:
$_GET = []```

**Comment by engram-design** [comment]
@pascalminator Just wanted to confirm your kits settings?

<img width="1151" alt="image" src="https://github.com/user-attachments/assets/33b3ae8a-6dee-4bbd-b223-84166b62969b">

@siebird A [502 Bad Gateway](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/502) might be unrelated to this one. Can you clear Craft caches and try again?

**Comment by pascalminator** [comment]
Here you go for my kit settings:

![config-fa-kit](https://github.com/user-attachments/assets/417e9c26-77ca-4d15-91ca-d1a51a4f22e5)

**Comment by engram-design** [comment]
Looks all correct to me, thanks for sending through. I have just pushed a fix for this error, but more to the point - I'm not _quite_ sure why it's reporting back as no icons, so this might not fix the underlying issue.

To get this early, run `composer require verbb/icon-picker:"dev-craft-5 as 3.0.1"`.

**Comment by d-karstens** [comment]
We have been running into the same 502 error issue as well, have tried with the above fix and it didn't seem to help

```
2024-11-26 14:45:18 [ERROR] FontAwesome Kit API error: “error code: 502” /app/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:111
2024-11-26 14:45:18 [INFO] Request context:
$_GET = []

$_POST = []
```

**Comment by d-karstens** [comment]
Unsure if it's relevant, but we have 2 staging environments and a local - the error occurs on our local environment and one of the staging environments, but everything works normally on the other staging

They're all using the same API key

**Comment by d-karstens** [comment]
We managed to figure it out from there, generating a new API key for each environment solved it 👍

**Comment by engram-design** [comment]
Yeah, that fix would've just "got out of the way" for any request-related errors, which the 502 error was the underlying one.

Good to know about the API key, maybe there were limits being hit?

**Comment by engram-design** [comment]
Updated in [3.0.3](https://github.com/verbb/icon-picker/releases/tag/3.0.3)

**Comment by AmarHuda1** [comment]
Hi @engram-design, I am also having the same problem with Icon Picker v2.0.23 in Craft CMS 4.

**Comment by engram-design** [comment]
Fixed for the next release. To get this early, run `composer require verbb/icon-picker:"dev-craft-4 as 2.0.23"`.
